rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,210 @@
+storage.
+
+### Use an Externally Maintained Asset ID Reservation System
+
+Neither option presented seems to handle the reserving of asset IDs. Therefore, 
+neither of these will fully replace currency factory and would more correctly 
+replace our own pallet-assets. A replacement for pallet-assets may be a good 
+idea but would require much more substantial changes.
+
+* **Parity's pallet-assets**
+  
+  Forking this pallet and maintaining extra features we need (hold) is an 
+  option.
+  
+  > It is not clear from RFC why substrate assets are good. IMHO it should be 
+  clarified - year of man work to make UI/QA/FE/users happy is what we lack. 
+  IMHO better to have fork of assets and collaborate with ecosystem to tune 
+  wallets. Do not write own wallets.
+  
+  #### Pros
+  
+  * Very nice UI compatibility with Polkadot Dashboard
+
+  * Maintained by Parity
+
+  * Simple Interface
+
+  * Can run two pallets at same time for two purposes (external and mintable) 
+  (Demonstrated by Moonbeam)
+
+  * Handles WASM contracts addresses as IDs
+
+  * Permissionless when needed
+
+  * Supported by Polkadot wallets
+  
+  * Supports asset metadata
+  
+  #### Cons
+  
+  * No asset ID reservation system, only prevents duplicates
+
+  * Hold functionality is not present (We depend on this in several cases)
+
+* **Moonbeam's pallet-asset-manager**
+
+Note: Not looking at using or forking this directly, but instead how they use 
+different instances of the same pallet for mintable and external assets.
+
+  #### Pros
+  
+  * Maintained by moonbeam
+  
+  #### Cons
+  
+  * Partially specialized
+
+  * No asset ID reservation system, only prevents duplicates
+
+#### Consequences
+
+* Any of these options will require a difficult runtime migration as we move 
+existing storage for pallet-assets/pallet-asset-registry/pallet-currency-factory 
+to an alternative.
+
+## Conclusion
+
+After analyzing possible solutions, the best course of action to deprecate 
+Currency Factory is to maintain a fork of Parity's pallet-assets, embrace the 
+design Moonbeam uses to maintain two instances of their pallet, and to remove 
+our asset ID reservation system (for release two).
+
+This can be accomplished in two main phases:
+
+1. Create a fork of Parity's pallet-assets and add the features we need
+
+2. Replace our pallet-assets and pallet-currency-factory with two instances of
+Parity's pallet-assets.
+
+Parts of either of these phases can be scaled down for release two.
+
+### Fork of Parity's pallet-assets
+
+1. Create a non-traditional fork of Parity's pallet-assets
+
+2. Implement `frame_support::traits::tokens::fungibles::MutateHold` for 
+pallet-assets
+
+3. Translate calls into our old pallet-assets and pallet-currency-factory to 
+our new pallet-assets
+
+### Manage Two Instances of pallet-assets
+
+1. Configure instance for mintable assets
+
+2. Configure instance for external assets
+
+3. Route calls between the two instances
```

Comment:
Looking for comments on this.
I was considering using Assets Registry as a routing layer, but that may be out of scope for Assets Registry.